installer/pkg/components/proxy/objects.go

```diff
@@ -5,30 +5,37 @@
 package proxy
 
 import (
+	"fmt"
+
 	"github.com/gitpod-io/gitpod/installer/pkg/common"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
 var Objects = common.CompositeRenderFunc(
 	configmap,
 	deployment,
 	networkpolicy,
 	rolebinding,
-	common.GenerateService(Component, map[string]common.ServicePort{
-		ContainerHTTPName: {
-			ContainerPort: ContainerHTTPPort,
-			ServicePort:   ContainerHTTPPort,
-		},
-		ContainerHTTPSName: {
-			ContainerPort: ContainerHTTPSPort,
-			ServicePort:   ContainerHTTPSPort,
-		},
-		MetricsContainerName: {
-			ContainerPort: PrometheusPort,
-			ServicePort:   PrometheusPort,
-		},
-	}, func(spec *corev1.ServiceSpec) {
-		spec.Type = corev1.ServiceTypeLoadBalancer
-	}),
+	func(cfg *common.RenderContext) ([]runtime.Object, error) {
+		return common.GenerateService(Component, map[string]common.ServicePort{
+			ContainerHTTPName: {
+				ContainerPort: ContainerHTTPPort,
+				ServicePort:   ContainerHTTPPort,
+			},
+			ContainerHTTPSName: {
+				ContainerPort: ContainerHTTPSPort,
+				ServicePort:   ContainerHTTPSPort,
+			},
+			MetricsContainerName: {
+				ContainerPort: PrometheusPort,
+				ServicePort:   PrometheusPort,
+			},
+		}, func(service *corev1.Service) {
+			service.Spec.Type = corev1.ServiceTypeLoadBalancer
+			service.Annotations["external-dns.alpha.kubernetes.io/hostname"] = fmt.Sprintf("%s,*.%s,*.ws.%s", cfg.Config.Domain, cfg.Config.Domain, cfg.Config.Domain)
```

Comment:
What happens if they're using something other than external-dns?

Reply:
If they're not using external-dns, nothing.
If they use something other than the annotation, I'd propose we cross that bridge when we get there. There's a good chance we'll need annotations on that particular service, and in that case we could add that back in for that particular case.

I'm afraid that providing something as generalised as a `component` section, we make it too easy for ourselves to open the config surface for too many potentialities thereby inadvertently letting it grow too big (again).

Reply:
Pragmatic and fair enough.